File: java/src/test/java/com/alxarafe/domain/topology/entity/AisleTest.java
package com.alxarafe.domain.topology.entity;

import com.alxarafe.app.domain.topology.entity.Aisle;
import com.alxarafe.app.domain.topology.valueobject.AisleCode;
import com.alxarafe.app.domain.topology.valueobject.AisleId;
import com.alxarafe.app.domain.topology.valueobject.ZoneId;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class AisleTest {

    @Test
    void create() {
        var id = new AisleId("018e4e3a-3e7b-7b3e-8000-000000000001");
        var zoneId = new ZoneId("018e4e3a-3e7b-7b3e-8000-000000000002");
        var code = new AisleCode("A-01");
        var aisle = new Aisle(id, zoneId, code);

        assertEquals(id, aisle.id());
        assertEquals(zoneId, aisle.zoneId());
        assertEquals(code, aisle.code());
        assertFalse(aisle.isBlocked());
    }

    @Test
    void block() {
        var aisle = new Aisle(
                new AisleId("018e4e3a-3e7b-7b3e-8000-000000000001"),
                new ZoneId("018e4e3a-3e7b-7b3e-8000-000000000002"),
                new AisleCode("A-01"));
        aisle.block();
        assertTrue(aisle.isBlocked());
    }

    @Test
    void unblock() {
        var aisle = new Aisle(
                new AisleId("018e4e3a-3e7b-7b3e-8000-000000000001"),
                new ZoneId("018e4e3a-3e7b-7b3e-8000-000000000002"),
                new AisleCode("A-01"));
        aisle.block();
        aisle.unblock();
        assertFalse(aisle.isBlocked());
    }
}

File: php/tests/Unit/Domain/Topology/Entity/AisleTest.php
<?php

declare(strict_types=1);

namespace Tests\Unit\Domain\Topology\Entity;

use Alxarafe\App\Domain\Topology\Entity\Aisle;
use Alxarafe\App\Domain\Topology\ValueObject\AisleCode;
use Alxarafe\App\Domain\Topology\ValueObject\AisleId;
use Alxarafe\App\Domain\Topology\ValueObject\ZoneId;
use PHPUnit\Framework\TestCase;

final class AisleTest extends TestCase
{
    public function testCreate(): void
    {
        $id = new AisleId('018e4e3a-3e7b-7b3e-8000-000000000001');
        $zoneId = new ZoneId('018e4e3a-3e7b-7b3e-8000-000000000002');
        $code = new AisleCode('A-01');
        $aisle = new Aisle($id, $zoneId, $code);

        self::assertTrue($id->equals($aisle->id()));
        self::assertTrue($zoneId->equals($aisle->zoneId()));
        self::assertTrue($code->equals($aisle->code()));
        self::assertFalse($aisle->isBlocked());
    }

    public function testBlock(): void
    {
        $aisle = new Aisle(
            new AisleId('018e4e3a-3e7b-7b3e-8000-000000000001'),
            new ZoneId('018e4e3a-3e7b-7b3e-8000-000000000002'),
            new AisleCode('A-01'),
        );
        $aisle->block();
        self::assertTrue($aisle->isBlocked());
    }

    public function testUnblock(): void
    {
        $aisle = new Aisle(
            new AisleId('018e4e3a-3e7b-7b3e-8000-000000000001'),
            new ZoneId('018e4e3a-3e7b-7b3e-8000-000000000002'),
            new AisleCode('A-01'),
        );
        $aisle->block();
        $aisle->unblock();
        self::assertFalse($aisle->isBlocked());
    }
}
